### ISSUE TITLE
Feature Request: make blurradius animatable

### ISSUE THREAD

**Issue by getaaron** [evadne/LiveFrost#9]
I think this would require a significant rewrite.  This is outside my area of expertise, so I'm not sure what the feasibility is.  I think we'd have to extend CALayer instead of UIView (see http://stackoverflow.com/questions/2395382/animating-a-custom-property-of-calayer-subclass).

**Comment by evadne** [comment]
I don’t think this will require a heavy rewrite. Note that what CAAnimation
all does is to solve value at time. It is IMHO possible to — duh, exactly
what your link mentioned. We can use a custom subclass and stash blur
radius (kernel size) on it. I am not sure if it will be fast enough though.

On Nov 20, 2013, at 6:09 AM, Aaron Brager notifications@github.com wrote:

I think this would require a significant rewrite. This is outside my area
of expertise, so I'm not sure what the feasibility is. I think we'd have to
extend CALayer instead of UIView (see
http://stackoverflow.com/questions/2395382/animating-a-custom-property-of-calayer-subclass
).

—
Reply to this email directly or view it on
GitHubhttps://github.com/radi/LiveFrost/issues/9
.

**Comment by ghost** [comment]
What @evadne said, and we'd also have to override +layerClass on the UIView to return the class of the new CALayer subclass.

I'd like to avoid changing the existing interface, since nobody wants to update their project submodules and pods and — SURPRISE! — experience the fun and excitement of seeing their build break.

**Comment by ghost** [comment]
For what it's worth, cross fading between blurred images does a really good job of simulating an animated blur radius. I'm normally pretty particular about how animations look, but even I had a hard time telling that the blur radius wasn't animating.

**Comment by ArtFeel** [comment]
I agree with @LucasTizma, in my project I'm simply fadeIn view.alpha, and it looks very good.

**Comment by getaaron** [comment]
The alpha fade looks good on some views, but it doesn't work as well on high contrast / sparsely populated views.

**Comment by ghost** [comment]
To add my two cents, I do think that migrating to a CALayer subclass with a UIView subclass for an interface is the best long term solution. That should yield a few more benefits, like allowing us to keep the entire blur effect in lockstep with animation.

It's not quite as high on the list of priorities right now, but that would probably be our next major release, `2.0` or something like that.

As for `blurradius`, we do enough pre-caching that making `blurradius` animatable will probably be really, really slow. But I think it's still worth trying.

If it's too slow, there will just have to be another barrier in the interface to make sure that $DEVELOPER really, really wants to do this.

Kind of like [trying to die in a LucasArts game](https://www.youtube.com/watch?v=1-_-QxerYno).

**Comment by getaaron** [comment]
Yeah. Maybe the animated `blurRadius` Is only enabled if you win a round of Insult Swordfighting.

**Comment by whitneyland** [comment]
Another thing that would make this nice is that the built in iOS8 UIBlurEffect still does not have an animatable radius.  Search continues...
